### ISSUE TITLE
[BUG]: ACT test failure for aria-checked when set to "undefined" 

### ISSUE THREAD

**Issue by tombrunet** [IBMa/equal-access#1270]
### Project

a11y checker engine

### Browser

Chrome

### Operating system

MacOS

### Description

An ACT example is as following: 
https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/proposed/
> The aria-checked attribute does not have a value.
> 
> Note: The HTML validator flags an aria-checked attribute with an empty value as an issue. However, since WAI-ARIA 1.1, aria-checked has a default value of undefined.
> 
```
<div role="checkbox" aria-checked>Accept terms and conditions</div>
```

We flag this as an issue.

### Steps to reproduce

See above.

**Comment by philljenkins** [comment]
It seems to me that an undefined value is not correct.  I do not know when a checkbox is undefined, shouldn't it either be checked or unchecked? 
Is this for some scripting or loading situation where the initial value is undefined (temporarily), so it doesn't get processed, and then based on some user action, loading, or other programmatic action it gets checked (or unchecked) and then processed correctly?

ARIA 1.0 https://www.w3.org/TR/wai-aria-1.0/states_and_properties
> ... true/false: Value representing either true or false, with a default "false" value.

ARIA 1.1 https://www.w3.org/TR/wai-aria/#requiredState
> ... Content authors MUST provide a non-empty value for **_required_** states and properties...
> ... Content authors MAY provide values for **_supported_** states and properties, but need not in some cases where default values are sufficient.
> ... When a widget is missing required owned elements due to script execution or loading, authors MUST mark a containing element with [aria-busy](https://www.w3.org/TR/wai-aria/#aria-busy) equal to true. For example, until a page is fully initialized and complete, an author could mark the document element as busy.

https://www.w3.org/TR/wai-aria/#checkbox
> see the screenshot below of the `checkbox(role)` in the spec. `aria-checked` is defined as **_"Required"_** 

Are you saying that this would be a valid example after loading?
`<li role="menuitemcheckbox" aria-checked>Not sorted yet</li>`

Seems to me it would still be "mo better" to be set to false (unchecked) instead of undefined. And my reading of the spec says it's a required state. Perhaps at best its a temporary false positive while loading or scripting is being processed.

![image](https://user-images.githubusercontent.com/16193609/214344639-f638195a-fcf3-4fa5-b13f-b5e76e1cc5d3.png)

**Comment by philljenkins** [comment]
Just read later in the ARIA 1.1 spec that the **_aira-checked (state)_** says:
https://www.w3.org/TR/wai-aria/#aria-checked

> Values:
> ...
> undefined (default): The element does not support being checked.

This also doesn't make sense to me when the element is `<checkbox>` which does support it.

**Comment by philljenkins** [comment]
The ACT example mentioned above is
> Proposed ARIA state or property has valid value 
> https://wai-wcag-act-rules.netlify.app/standards-guidelines/act/rules/6a7281/proposed/
> ...
> Inapplicable Example 3

I recommend commenting on the proposed rule (and clarifying the spec).

**Comment by philljenkins** [comment]
I believe the ACT code example doesn't make valid sense from a legal point of view
`<div role="checkbox" aria-checked>Accept terms and conditions</div>`
Wouldn't they want the default to be "unchecked" and not processed until the user "checks" (accepts) the **_Terms and Conditions_**?

**Comment by shunguoy** [comment]
https://github.com/w3c/aria/issues/1622

**Comment by shunguoy** [comment]
per team discussion, the fix will be limited to (1) ignoring the check if the element is hidden; and (2) removing duplicated attributes in the reporting.

**Comment by ErickRenteria** [comment]
I'm not getting a different result with the new rule vs the old: @shunguoy @philljenkins Please review
<img width="932" alt="Screen Shot 2023-02-06 at 1 04 31 PM" src="https://user-images.githubusercontent.com/62436670/217063161-d0d3eef5-db7c-46c9-b3e6-a1d14298cd23.png">

**Comment by philljenkins** [comment]
@ErickRenteria when running this new rule against the [ACT 6a7281 test case](https://www.w3.org/WAI/standards-guidelines/act/rules/6a7281/proposed/#test-cases), it should not report an error as it use for:
Open in a new tab and  test
[Inapplicable Example 3](https://w3.org/WAI/content-assets/wcag-act-rules/testcases/6a7281/d82c477ac58becfda4233237a642b8ef70eb6c5f.html)

**Comment by ErickRenteria** [comment]
Validated
